src/LDLFactorizations.jl

```diff
@@ -192,6 +198,13 @@ function ldl_numeric_upper!(n, Ap, Ai, Ax, Cp, Ci, Lp, parent, Lnz, Li, Lx, D, Y
       Lnz[i] += 1
       top += 1
     end
+    if dynamic_regul && abs(D[k]) < dynamic_dict[:tol]
+      if P[k] <= dynamic_dict[:n_d]
+        D[k] += dynamic_dict[:r1]
```

Comment:
Also, shouldn't this be something like
```
r1 = dynamic_dict[:r1]
D[k] = sign(r1) * max(abs(D[k]) + abs(r1), abs(r1))
```
Suppose `r1 = 1.0e-8`, `tol = 1.0e-8` but `D[k] = -1.0e-8`. Then, you're setting it to zero.